src/Database/Connection.php

```diff
@@ -44,6 +44,20 @@ class Connection implements ConnectionInterface
 {
     use TypeConverterTrait;
 
+    /**
+     * Primary/write connection role name
+     *
+     * @var string
+     */
+    public const ROLE_PRIMARY = 'primary';
```

Comment:
I am wondering is `write/read` would be more newbie friendly than `primary/replica`.

Reply:
I was following mark's nomenclature. I originally thought read/write would work too.

Databases say `replica` (not primary though), but we're writing this for users not DBAs.

Reply:
I used primary/replica in the issue as that is the terminology that databases generally use as well. To me 'read/write' implies that I cannot read from the primary which is not true, and in several common scenarios one would need to read from the primary.

Reply:
That is true. We could have `primary` and `read`. Calling it `write` does imply you're not allowed to read from it.